cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillRunnerService.java

```diff
@@ -252,6 +316,378 @@ public void start() {
     }
   }
 
+  @Override
+  public void onStart(Collection<NamespaceDetail> namespaceDetails) {
+    for (NamespaceDetail namespaceDetail : namespaceDetails) {
+      String namespace = namespaceDetail.getProperties().get(KubeMasterEnvironment.NAMESPACE_PROPERTY);
+      // Start job watchers on non-system k8s namespaces. We add watchers on the system namespace in start()
+      if (namespace != null && !resourceWatchers.containsKey(namespace) && !namespace.equals(kubeNamespace)) {
+        addAndStartJobWatcher(namespace);
```

Comment:
should make this method synchronized now that different threads can be calling methods that try to modify this map.

Reply:
it should also be private now instead of public

Reply:
done